app/(root)/profile/edit/page.tsx

```diff
@@ -0,0 +1,14 @@
+import { auth } from "@/auth";
+import EditProfile from "@/components/profile/EditProfile";
+import { getUser } from "@/lib/actions/user.actions";
+import { User } from "@prisma/client";
+
+const EditProfileWrapper = async () => {
+  const session = await auth();
+  const userEmail = session && (await session.user?.email);
+  const user = userEmail && (await getUser(userEmail));
+
+  return <EditProfile user={user as User} />;
```

Comment:
Will there ever be a use case where this user is undefined?  Should I handle that error here, or in EditProfile?

Reply:
@DDVVPP Ideally you don't let someone (because of your middleware) access this page without a session. So it shouldn't happen. 

If you want to handle the potential error- you should see what your user returns. Your server-action currently returns an "error" key in the object if there's no user.

```js
  } catch (error) {
    console.error("Error finding user:", error);
    return { error: "User not found!" };
  }
 ```
 
 You could check to see if the error exists and then either redirect them or send them to a 'notFound' page:
 https://nextjs.org/docs/app/api-reference/functions/not-found

Reply:
Okay, in that case I'll leave it be